sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -0,0 +1,505 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_iot_hub_client.h
+ *
+ * @brief definition for Azure IoT Hub device-client SDK.
+ */
+
+#ifndef _az_IOT_HUB_CLIENT_H
+#define _az_IOT_HUB_CLIENT_H
+
+#include <az_result.h>
+#include <az_span.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * @brief Azure IoT Hub Client options.
+ *
+ */
+typedef struct az_iot_hub_client_options
+{
+  az_span module_id; /** @brief The module name (if a module identity is used). */
+  az_span user_agent; /** @brief The user-agent is a formatted string that will be used for Azure

```

Comment:
>/** @brief The user-agent is a formatted string that will be used for Azure [](start = 22, length = 75)

Could we (or should we?) put the doxygen comments on a separate line as the actual variable itself?  To my eye this feels more readable.

Reply:
The main problem I have with this is the following:
The `.clang_format` we currently have limits indentations to 2 spaces instead of four. With the documentation on it's own line, it becomes somewhat difficult as the vertical space the struct declaration takes up expands to track where the struct starts and ends. There isn't a big difference between the left justified alignment on the beginning and two spaces in for each member. Some members start to look like global variable declarations. This at least tries to keep it compact.

Reply:
1. I kind of agree with @jspaith _if_ we can move them outside of the struct. 
1. @danewalton made a comment above about using > instead of @brief ,etc. That should make things a bit more compact.

I'll defer to @danewalton since he's the doc expert. What do you think about using: https://stackoverflow.com/questions/7199151/use-doxygen-to-document-members-of-a-c-structure-outside-of-the-structure-defini ?

```
/** @struct foreignstruct
 *  @brief This structure blah blah blah...
 *  @var foreignstruct::a 
 *  Member 'a' contains...
 *  @var foreignstruct::b 
 *  Member 'b' contains...
 */
```

Azure SDKs - what's your take on this? I couldn't find any sample of a public struct member doxygen documentation.

Reply:
I moved them to the way @danewalton recommended for now. Please open a separate issue if we want to change again.